docs: reformatting tables in drivers/gm.adoc (smoe:docs_drivers_gm_tables_reformatting)

smoe opened the pull request:
These changes are mostly addressing the readability of this document. We should not have blocks when all we want is a readable sentence, and we should not have blocks when what we really want is just a table.

I was also a bit curious about how po4a will deal with those asciidoc table style operators. The table fields that present pin names are now preceeded with an "m" for monospace or "l" for being treated like a block.

smoe commented:
Rebased to master after conflicts.
The reformatting avoids blanks at the beginning of the lines that irritate the automated translators.

smoe commented:
Reformatting to help the automated translation may be a thing for the style guide.

silopolis commented:
#1530 style guide

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -740,24 +717,24 @@ Every RS485 module pin and parameter name begins as follows:
-# .in-0    - Select the first digital input module
-----
+|===
+m|gm.0.rs485.0.in-0 |First input of the node.
+2+|constructed from
```
I would exclude this from the table, this looks more like a formatting error  :innocent: 
Also ff.

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -301,25 +296,25 @@ gm.<nr of card>.write
 |===
 | Parameters         | Type and Read/Write | Parameter description
-| .step-type         | (u32, R/W)          | When 0, module produces Step/Dir signal. When 1, it
+l| .step-type         | (u32, R/W)          | When 0, module produces Step/Dir signal. When 1, it
```
```suggestion
m| .step-type         | (u32, R/W)          | When 0, module produces Step/Dir signal. When 1, it
```
"l" produces this ugly space...

smoe replied:
I have addressed these all locally.

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -189,71 +181,80 @@ gm.<nr of card>.read
-setp gm.0.encoder.0.index-invert 0
-setp gm.0.encoder.0.min-speed-estimate 0.1 # in position unit/s
-setp gm.0.encoder.0.position-scale 20000   # 10 encoder rev cause the machine to
+[options="header"]
+|===
+| setp pin seetting                          | Comment
+l|setp gm.0.encoder.0.counter-mode 0         | 0: quad, 1: stepDir
+l|setp gm.0.encoder.0.index-mode 1           | 0: reset pos at index, 1:round pos at index
+l|setp gm.0.encoder.0.counts-per-rev 2000    | GM process encoder in 4x mode, 4x500=2000
+l|setp gm.0.encoder.0.index-invert 0         |
+l|setp gm.0.encoder.0.min-speed-estimate 0.1 | in position unit/s
+l|setp gm.0.encoder.0.position-scale 20000   | 10 encoder rev cause the machine to
```
Formatting this as table looses the connection that it is a code example

smoe replied:
done

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -331,20 +326,20 @@ image::images/GM_RefSignals.png["Reference signal timing diagrams",align="center
-setp gm.0.stepgen.0.stepspace1000       # 1000 ns = 1 µs
-setp gm.0.stepgen.0.dirdelay 2000       # 2000 ns = 2 µs
-----
+[options="header"]
+|===
+| setp pin seetting                  | Comment
+l|setp gm.0.stepgen.0.step-type 0    | 0:stepDir, 1:UpDown, 2:Quad
+l|setp gm.0.stepgen.0.control-type 0 | 0:Pos. control, 1:Vel. Control
+l|setp gm.0.stepgen.0.invert-step1 0 |
+l|setp gm.0.stepgen.0.invert-step2 0 |
+l|setp gm.0.stepgen.0.maxvel 0       |do not set maxvel for step lenerator, let interpolator control it.
+l|setp gm.0.stepgen.0.maxaccel 0     |do not set max acceleration for step generator, let interpolator control it.
+l|setp gm.0.stepgen.0.position-scale 1000 |1000 step/position unit
+l|setp gm.0.stepgen.0.steplen 1000   | 1000 ns = 1 µs
+l|setp gm.0.stepgen.0.stepspace 1000 | 1000 ns = 1 µs
+l|setp gm.0.stepgen.0.dirdelay 2000  | 2000 ns = 2 µs
+|===
```
Formatting this as table looses the connection that it is a code example. I would rather add `[source,{hal}]` to the code block.

smoe replied:
Hm. Only the left side is executable, not the comment on the right.

hansu replied:
>Agree with @hansu so same comment and proposal as above.
"Right side" could be converted to # <n> callouts but not convinced it would be beneficial here.

👍

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -370,13 +363,12 @@ gm.<nr of card>.read
-  then power enable pins of axis- and CAN connectors
-  are set to high, otherwise set to low.
+then power enable pins of axis- and CAN connectors are set to high, otherwise set to low.
 | gm.<nr of card>.power-fault  | (bit, Out)         | Power fault input.
```
```suggestion
| gm.<card_no>.power-fault  | (bit, Out)         | Power fault input.
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -391,13 +383,9 @@ Every axis DAC pin and parameter name begins as follows:
-where _<nr of axis>_ is from 0 to 5. For example:
-
-----
-gm.0.dac.0.value
-----
-
+where _<nr of axis>_ is from 0 to 5. For example, `gm.0.dac.0.value`
 refers to the output voltage of DAC module of axis 0.
+
```
```suggestion
gm.<card_no>.dac.<axis_no>
----

Where _<axis_no>_ is from 0 to 5. +
For example:

.First DAC module output voltage on card 0
[source,{hal}]
----
gm.0.dac.0.value
----

```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -455,13 +438,8 @@ Every CAN pin and parameter name begins as follows:
-
-----
-gm.0.can-gm.0.position
-----
-
-refers to the output position of axis 0 in position units.
+where _<nr of axis>_ is from 0 to 5.
+For example, `gm.0.can-gm.0.position` refers to the output position of axis 0 in position units.
```
```suggestion
gm.<card_no>.can-gm.<axis_no>
----

Where _<axis_no>_ is from 0 to 5. +
For example:

.Output position in position units on Axis 0 on card 0
[source,{hal}]
----
gm.0.can-gm.0.position
----
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -511,17 +489,17 @@ Watchdog timer overrun causes the set of power-enable to low in hardware.
-  gm._<nr of card>_.read function must be executed. The gm._<nr of card>_.read
-  is typically added to servo-thread, so watch timeout is typically set to 3 times
-  of the servo period.
+| gm._<nr of card>_.watchdog-enable     | (bit, R/W)         |
```
```suggestion
| gm._<card_no>_.watchdog-enable     | (bit, R/W)         |
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -511,17 +489,17 @@ Watchdog timer overrun causes the set of power-enable to low in hardware.
+Enable watchdog timer.
+It is strongly recommended to enable watchdog timer,
+because it can disables all the servo amplifiers by pulling down all enable signal in case of PC error.
+| gm._<nr of card>_.watchdog-timeout-ns | (float, R/W)       |
```
```suggestion
| gm._<card_no>_.watchdog-timeout-ns | (float, R/W)       |
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -511,17 +489,17 @@ Watchdog timer overrun causes the set of power-enable to low in hardware.
+It is strongly recommended to enable watchdog timer,
+because it can disables all the servo amplifiers by pulling down all enable signal in case of PC error.
+| gm._<nr of card>_.watchdog-timeout-ns | (float, R/W)       |
+Time interval in within the gm._<nr of card>_.read function must be executed.
+The gm._<nr of card>_.read is typically added to servo-thread,
+so watch timeout is typically set to 3 times of the servo period.
```
```suggestion
Time interval in within the gm._<card_no>_.read function must be executed.
The gm._<card_no>_.read is typically added to servo-thread,
so watch timeout is typically set to 3 times of the servo period.
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -548,12 +526,8 @@ The GM6-PCI motion control card has two limit- and one homing switch input for e
-where _<nr of axis>_ is from 0 to 5. For example:
-
-----
-gm.0.joint.0.home-sw-in
-----
-
+where _<nr of axis>_ is from 0 to 5. For example,
+`gm.0.joint.0.home-sw-in`
 indicates the state of the axis 0 home switch.
```
```suggestion
gm.<card_no>.joint.<axis_no>
----

Where _<axis_no>_ is from 0 to 5. +
For example:

.Home switch state on Axis 0 on card 0
[source,{hal}]
----
gm.0.joint.0.home-sw-in
----

```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -702,25 +676,24 @@ where <module ID> is from 00 to 15.
-gm.0                 # Means the first GM6-PCI motion control card (PCI card address = 0)
-.rs485.0             # Select node with address 0 on the RS485 bus
-.relay-0             # Select the first relay
-----
+|===
+m|gm.0.rs485.0.relay-0 | First relay of the node.
+m|gm.0                 | Means the first GM6-PCI motion control card (PCI card address = 0)
+m|.rs485.0             | Select node with address 0 on the RS485 bus
+m|.relay-0             | Select the first relay
+|===
```
```suggestion
.First relay on RS485 node 0 on card 0
[source,{hal}]
----
gm.0.rs485.0.relay-0 # First relay of the node.
                     # gm.0 - Means the first GM6-PCI motion control card (PCI card address = 0)
                     # .rs485.0 - Select node with address 0 on the RS485 bus
                     # .relay-0 - Select the first relay
----
```

hansu replied:
```
.First relay on RS485 node 0 on card 0
====
[source,{hal}]
----
gm.0.rs485.0.relay-0 
----
`gm.0` Means the first GM6-PCI motion control card (PCI card address = 0)
`.rs485.0` Select node with address 0 on the RS485 bus
`.relay-0` Select the first relay
====
```
I agree, maybe the explanation doesn't have to be in the code block.

But if we have it inside the code block, we can have it like this:
```
.First relay on RS485 node 0 on card 0
====
[source,{hal}]
----
  gm.0.rs485.0.relay-0  # First relay of the node.
# gm.0                    Means the first GM6-PCI motion control card (PCI card address = 0)
#     .rs485.0            Select node with address 0 on the RS485 bus
#             .relay-0    Select the first relay
----
```

silopolis replied:
> ```
> .First relay on RS485 node 0 on card 0
> ====
> [source,{hal}]
> ----
> gm.0.rs485.0.relay-0 
> ----
> `gm.0` Means the first GM6-PCI motion control card (PCI card address = 0)
> `.rs485.0` Select node with address 0 on the RS485 bus
> `.relay-0` Select the first relay
> ====
> ```
> 
> I agree, maybe the explanation doesn't have to be in the code block.

This may well be my preferred format :+1: , just using a `#<1><2><3>`  callouts or a simple unordered list for the 3 elements.

> But if we have it inside the code block, we can have it like this:
> 
> ```
> .First relay on RS485 node 0 on card 0
> ====
> [source,{hal}]
> ----
>   gm.0.rs485.0.relay-0  # First relay of the node.
> # gm.0                    Means the first GM6-PCI motion control card (PCI card address = 0)
> #     .rs485.0            Select node with address 0 on the RS485 bus
> #             .relay-0    Select the first relay
> ----
> ```

That's a clever one but my eyes on first sight so...

silopolis replied:
Thought about it again last night and I'm afraid that the example block makes it standout even more than the sibling without it.
Kind of a pity because it is semantically sound. Could we make something with a pinch of CSS Fu ? Like removing bold face for example block titles ? And left side border ?

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -740,24 +713,28 @@ Every RS485 module pin and parameter name begins as follows:
 gm.<nr of card>.rs485.<module ID>
 ----
 
-where <module ID> is from 00 to 15.
+where _<module ID>_ is from 00 to 15.
```
```suggestion
gm.<card_no>.rs485.<module_id>
----

Where _<module_id>_ is from 00 to 15.
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -777,39 +754,43 @@ Every RS485 module pin and parameter name begins as follows:
 gm.<nr of card>.rs485.<module ID>
 ----
 
-where <module ID> is from 00 to 15.
+where _<module ID>_ is from 00 to 15.
```
```suggestion
gm.<card_no>.rs485.<module_id>
----

Where _<module_id>_ is from 00 to 15.
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -740,24 +713,28 @@ Every RS485 module pin and parameter name begins as follows:
-# gm.0     - Means the first GM6-PCI motion control card (PCI card address = 0)
-# .rs485.0 - Select node with address 0 on the RS485 bus
-# .in-0    - Select the first digital input module
-----
+[cols="<3,<6"]
+|===
+m|gm.0.rs485.0.in-0 |First input of the node.
+|===
+constructed from
+[cols="<3,<6"]
+|===
+m|gm.0     | Means the first GM6-PCI motion control card (PCI card address = 0)
+m|.rs485.0 | Select node with address 0 on the RS485 bus
+m|.in-0    |Select the first digital input module
+|===
```
```suggestion
.First digital input module on RS485 node 0 on card 0
[source,{hal}]
----
gm.0.rs485.0.in-0 # First input of the node.
                  # gm.0 - Means the first GM6-PCI motion control card (PCI card address = 0)
                  # .rs485.0 - Select node with address 0 on the RS485 bus
                  # .in-0 - Select the first digital input module
----
```

smoe replied:
Reversed towards something reasonably similar.

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -777,39 +754,43 @@ Every RS485 module pin and parameter name begins as follows:
-# gm.0     - Means the first GM6-PCI motion control card (PCI card address = 0)
-# .rs485.0 - Select node with address 0 on the RS485 bus
-# .adc-0   - Select the first analogue input of the module
-----
+[cols="<3,<6"]
+|===
+m|gm.0.rs485.0.adc-0 | First analogue channel of the node.
+|===
+constructed from
+[cols="<3,<6"]
+|===
+m|gm.0     | Means the first GM6-PCI motion control card (PCI card address = 0)
+m|.rs485.0 | Select node with address 0 on the RS485 bus
+m|.adc-0   | Select the first analogue input of the module
+|===
```
```suggestion
.First analog channel on RS485 node 0 on card 0
[source,{hal}]
----
gm.0.rs485.0.adc-0 # First analogue channel of the node.
                   # gm.0 - Means the first GM6-PCI motion control card (PCI card address = 0)
                   # .rs485.0 - Select node with address 0 on the RS485 bus
                   # .adc-0 - Select the first analogue input of the module
----
```

smoe replied:
The original was better, I think.

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -189,71 +181,79 @@ gm.<nr of card>.read
+l|setp gm.0.encoder.0.position-scale 20000   | 10 encoder rev cause the machine to move one position unit (10x2000)
+|===
 
 Connect encoder position to LinuxCNC position feedback:
 
```
```suggestion
.Connect encoder position to LinuxCNC joint position feedback
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -274,13 +274,7 @@ Every StepGen pin and parameter name begins as follows:
 gm.<nr of card>.stepgen.<nr of axis>
```
```suggestion
gm.<card_no>.stepgen.<axis_no>
```

silopolis commented on docs/src/drivers/gm.adoc:
```diff
@@ -834,35 +815,39 @@ Note that on the Teach Pendant module it cannot be changed, and pre-programmed a
-# gm.0     - Means the first GM6-PCI motion control card (PCI card address = 0)
-# .rs485.0 - Select node with address 0 on the RS485 bus
-# .adc-0   - Select the first analogue input of the module
-----
+[cols="<3,<6"]
+|===
+| gm.0.rs485.0.adc-0 | First analogue channel of the node.
+|===
+constructed from
+[cols="<3,<6"]
+|===
+m| gm.0     | Means the first GM6-PCI motion control card (PCI card address = 0)
+m| .rs485.0 | Select node with address 0 on the RS485 bus
+m| .adc-0   | Select the first analogue input of the module
+|===
```
```suggestion
.First analog channel on RS485 node 0 on card 0
[source,{hal}]
----
gm.0.rs485.0.adc-0 # First analogue channel of the node.
                   # gm.0 - Means the first GM6-PCI motion control card (PCI card address = 0)
                   # .rs485.0 - Select node with address 0 on the RS485 bus
                   # .adc-0 - Select the first analogue input of the module
----

smoe replied:
Hm. I then prefer the original.

hansu replied:
What do you think of this: https://github.com/LinuxCNC/linuxcnc/pull/2037#discussion_r980201898?

silopolis replied:
> Hm. I then prefer the original.

Not having it indented makes it unobvious that these lines pertains to the first one because they don't differ much from another commented line.

smoe replied:
> #2037 (comment)

The indented one is very nice, indeed. A conditione sine qua non for me is that code-blocks can be copied in toto and are executable. So, the non-executable parts need to be commented "away". You seem to have addressed that, so I am fine with it.

smoe replied:
I think to have addressed this now.

silopolis reviewed (changes requested):
* It is common practice to have user replaceable content emphasized and `_<snake_cased>_` so I push it again here and will do in the style guide too.
* Re-introduced examples as code block for better structure, markup semantics, readability/scannability, and look.
* Added titles describing examples
* Made some rephrasing trying to avoid passive form

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -82,14 +82,11 @@ Every GPIO pin and parameter name begins as follows:
 ----
 
-where _<nr of gpio con>_ is from 0 to 3. For example:
+where _<nr of gpio con>_ is from 0 to 3.
+For example, `gm.0.gpio.0.in-0` indicates the state of the
+first pin of the first GPIO connector on the GM6-PCI card.
```
Yeah but this is not a code example, just a pin name.

silopolis replied:
hmm, not a bad point... But I have the feeling that for someone configuring a machine and having to dive into documentation, the examples are almost as important to identify quickly than the syntax pattern, all the more in the beginning when she needs it most.

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -82,14 +82,11 @@ Every GPIO pin and parameter name begins as follows:
 gm.<nr of card>.gpio.<nr of gpio con>
```
The underscores are not really needed here. Maybe even misleading as it look then more like a defined name or parameter.

silopolis replied:
Not needed per se, but as I said, it's current recommended style and also avoid suggesting spaces are allowed.
Plus, in this very case, this is somehow a defined value.

hansu replied:
Recommended by the Gitlab style guide?

silopolis replied:
Not explicitly but the example suggests it https://docs.gitlab.com/ee/development/documentation/styleguide/#placeholder-text

Same here http://courses.cms.caltech.edu/cs11/material/general/usage.html

And the very nice https://redhat-documentation.github.io/asciidoc-markup-conventions/ (parameters and variables at the bottom)

hansu replied:
Okay, persuaded :slightly_smiling_face:

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -189,71 +181,79 @@ gm.<nr of card>.read
+(non-quadrature) or step/dir signal sensor. When false, it
+counts in quadrature mode.
+l| .index-mode         | (bit, R/W)          |
+When True and .index-enabled is also true, .counts and
+.position are rounded (based on .counts-per-rev) at rising
```
```suggestion
When True and _.index-enabled_ is also true, _.counts_ and
_.position_ are rounded (based on _.counts-per-rev_) at rising
```
Maybe even highlight them.

silopolis replied:
These all should be put between backtick actually ! Missed that :/

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -189,71 +181,79 @@ gm.<nr of card>.read
-setp gm.0.encoder.0.index-mode 1           # 0: reset pos at index, 1:round pos at index
-setp gm.0.encoder.0.counts-per-rev 2000      # GM process encoder in 4x mode, 4x500=2000
-setp gm.0.encoder.0.index-invert 0
-setp gm.0.encoder.0.min-speed-estimate 0.1 # in position unit/s
-setp gm.0.encoder.0.position-scale 20000   # 10 encoder rev cause the machine to
-                                             move one position unit (10x2000)
-----
+[options="header"]
+|===
+| setp pin setting                           | Comment
+l|setp gm.0.encoder.0.counter-mode 0         | 0: quad, 1: stepDir
+l|setp gm.0.encoder.0.index-mode 1           | 0: reset pos at index, 1:round pos at index
+l|setp gm.0.encoder.0.counts-per-rev 2000    | GM process encoder in 4x mode, 4x500=2000
+l|setp gm.0.encoder.0.index-invert 0         |
+l|setp gm.0.encoder.0.min-speed-estimate 0.1 | in position unit/s
+l|setp gm.0.encoder.0.position-scale 20000   | 10 encoder rev cause the machine to move one position unit (10x2000)
+|===
```
I totally agree.

hansu commented:
Am 26.09.22 um 19:59 schrieb Jérémie Tarot:
>
> ***@***.**** commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In docs/src/drivers/gm.adoc 
> <https://github.com/LinuxCNC/linuxcnc/pull/2037#discussion_r980347332>:
>
> > @@ -82,14 +82,11 @@ Every GPIO pin and parameter name begins as follows:
>   gm.<nr of card>.gpio.<nr of gpio con>
>
> Not explicitly but the example suggests it 
> https://docs.gitlab.com/ee/development/documentation/styleguide/#placeholder-text
>
> Same here http://courses.cms.caltech.edu/cs11/material/general/usage.html
>
> And the very nice 
> https://redhat-documentation.github.io/asciidoc-markup-conventions/ 
> (parameters and variables at the bottom)
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/LinuxCNC/linuxcnc/pull/2037#discussion_r980347332>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AQGPJHBSDTOKVOVAPDL3JUTWAHP6JANCNFSM6AAAAAAQMKVLPU>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>
Okay, persuaded :slightly_smiling_face:

smoe commented:
Have rebased and pulled everything in. Please feel free to finalize as you see fit, maybe squash and upload.

smoe commented:
The build failure would be nice to see fixed, but that is beyond our control, I think.

smoe commented:
Followed your suggestions, rebased + squashed.

silopolis commented:
Le mer. 5 oct. 2022 à 15:36, Steffen Möller ***@***.***> a
écrit :

> Followed your suggestions, rebased + squashed.
>

 👍🙏

hansu commented:
> Followed your suggestions, rebased + squashed.

At least not here in this PR :(

smoe commented:
> > Followed your suggestions, rebased + squashed.
> 
> At least not here in this PR :(

Did I mess something up by force-pushing from different machines? Had a look at the diff and found some "l|" operators that I changed to "m" and some other "m|" where still missing. But wrt our discussion above the change back from the table to the verbatim text is in.

hansu commented:
I built it yesterday and found lots of the tabe issues and lots of code snippets which were formatted as table...

smoe commented:
> I built it yesterday and found lots of the table issues and lots of code snippets which were formatted as table...

I found one of those "constructed by" verbatim -> table rearragements, still, that I have now reversed. And I indeed had some "l" table operators, still, for whatever reason. Those have been "m"ed. But otherwise, from a look at the diff, I think these changes are decent.

hansu commented:
> But otherwise, from a look at the diff, I think these changes are decent.

I quick look on the diff looks good now 👍

silopolis commented:
Le mer. 14 sept. 2022 à 13:48, Steffen Möller ***@***.***> a
écrit :

> These changes are mostly addressing the readability of this document. We
> should not have blocks when all we want is a readable sentence, and we
> should not have blocks when what we really want is just a table.
>

Not sure I understand what you mean here 🤔

I was also a bit curious about how po4a will deal with those asciidoc table
> style operators. The table fields that present pin names are now preceeded
> with an "m" for monospace or "l" for being treated like a block.
>

It will be an interesting test for this matter indeed.
Nevertheless, I'm really not a fan of tables for such list and clearly
prefer them definition/labelled lists.

smoe commented on docs/src/drivers/gm.adoc:
```diff
@@ -167,96 +165,83 @@ RJ50 axis connector.
+m|setp gm.0.encoder.0.index-invert 0         |
+m|setp gm.0.encoder.0.min-speed-estimate 0.1 | in position unit/s
+m|setp gm.0.encoder.0.position-scale 20000   | 10 encoder rev cause the machine to move one position unit (10x2000)
+|===
```
done - thanks!

smoe commented:
Rebased and squashed against master.

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -686,40 +635,38 @@ http://www.generalmechatronics.com/data/products/robot_controller/PCI_UserManual
+#gm.0.rs485.0.relay-0  # First relay of the node.
+# gm.0  # Identifies the first GM6-PCI motion control card (PCI card address = 0)
+#    .rs485.0  # Selects node with address 0 on the RS485 bus
+#            .relay-0 # Selects the first relay
```
>The same for two further occurrences

hansu commented:
> The same width fot both tables looked better before:
> 
> ![Screenshot_2022-10-10_19-22-36](https://user-images.githubusercontent.com/67957916/194922217-66dd8572-0a45-4f50-8e65-670198ac8d57.png)
> 
> and here
> 
> ![Screenshot_2022-10-10_19-23-18](https://user-images.githubusercontent.com/67957916/194922311-d6f3d213-0cac-4a6a-b3a6-6fe7d92c3895.png)
> 
> Here is something missing ![Screenshot_2022-10-10_19-22-52](https://user-images.githubusercontent.com/67957916/194922361-95e75a61-b5b5-44f3-9ada-215a3d230310.png)

...

smoe commented:
The tables' widths is now consistently at 90% (before: 80% and 100%). Not sure about the "something missing".

hansu commented on docs/src/drivers/gm.adoc:
```diff
@@ -329,8 +307,8 @@ image::images/GM_RefSignals.png["Reference signal timing diagrams",align="center
-Setting StepGen module of axis 0 to generate 1000 step pulse per position unit:
-
+.Setting StepGen module of axis 0 to generate 1000 step pulse per position unit
+[source,{hal}]]
```
It's one closing bracket to much here
```suggestion
[source,{hal}]
```

hansu commented:
> Not sure about the "something missing".

I guess it's that: https://github.com/LinuxCNC/linuxcnc/pull/2037/files#r998451638

smoe commented:
> > Not sure about the "something missing".
> 
> I guess it's that: https://github.com/LinuxCNC/linuxcnc/pull/2037/files#r998451638

Judgeing from the diff it is fine. It only removes the empty lines.

smoe commented:
I just spotted that we are running into the Perl -> 5.36 transition on Debian unstable.

hansu commented:
Can you rebase this as a try to make the build passing?

smoe commented:
> Can you rebase this as a try to make the build passing?

@hansu  Worked!
To  not only keep our builders busy with the same thing again, I had added a few fixes on some random man pages.
